packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -297,7 +313,11 @@ class MenuContent extends React.Component {
           onKeyDown,
           isActive,
         });
-      // If the child has children then it is an item group, so iterate through it's children
+
+        if (hasHeaderTitle && item.props.subMenuItems && item.props.subMenuItems.length > 0) {
```

Comment:
Why should `hasHeaderTitle` decide whether or not there are sub menu items?

Reply:
If the user does not provide the `headerTitle` prop and the menu contains sub-menu-items, `hasSubmenuItem` will be true and trigger building a header and will create an empty `<div/>`. `hasHeaderTitle` makes sure to set `hasSubmenuItem` true only upon providing a valid `headerTitle` prop content.

Reply:
I don't think the `hasSubMenuItems` bool is needed, the `isSubMenu` bool below will render sub menu titles

Reply:
`isSubMenu` check is true only for second and higher-tier menus.

To render the header for main-menu(first-tier) we need the `hasSubMenuItems` check.

Reply:
> If the user does not provide the `headerTitle` prop and the menu contains sub-menu-items, `hasSubmenuItem` will be true and trigger building a header and will create an empty `<div/>`. `hasHeaderTitle` makes sure to set `hasSubmenuItem` true only upon providing a valid `headerTitle` prop content.

@lokesh-0813 ,  in this case we might have to add the prop level validation to trigger the warning mentioning that. Set the headertitle prop value as the submenutiems are present. and also update the doc mentioning that the the value is required when submenu items are present.

cc @neilpfeiffer

Reply:
+1 to @emilyrohrbough's comment and I still stand by my original comment. I think this logic could be simplified to not have new booleans - isn't it as simple as checking the value of  `this.props.headerTitle.length > 0` **AND** `item.props.subMenuItems && item.props.subMenuItems.length > 0` directly? If this condition evaluates to true, then build the header with title, otherwise do not build the header with title.

Reply:
@emilyrohrbough @nramamurth @ShettyAkarsh updated here b1254c430078f5a32a9c1de79d9402410ecc4df2